app/models/planning_application.rb

```diff
@@ -1029,7 +1029,11 @@ def set_key_dates
   end
 
   def application_type_determination_period
-    application_type.determination_period_days || DAYS_TO_EXPIRE
+    if pre_application?
+      local_authority.application_type_overrides.determination_period_days || DAYS_TO_EXPIRE
+    else
+      application_type.determination_period_days || DAYS_TO_EXPIRE
+    end
```

Comment:
Could we have a method on the application type instead?

I think we don't need to check whether it's pre-app or not. Just check if the override value has been set and if so, use this value but if not use the default value

Reply:
Yeah, e.g.:

```suggestion
      local_authority.application_type_overrides.determination_period_days ||
        application_type.determination_period_days ||
        DAYS_TO_EXPIRE
```

Reply:
Just need to consider when the application type overrides is something like

```
[
  {"code" => "preApp", "determination_period_days" => 30}, 
  {"code" => "HAPP", "determination_period_days" => 21}
]
```

Reply:
Speaking of, don't we want to store these overrides as an array of objects?
Unless i'm missing something